libgalois/include/katana/DynamicBitset.h

```diff
@@ -182,13 +182,14 @@ class KATANA_EXPORT DynamicBitset {
 
   /**
    * Check a bit to see if it is currently set.
-   * Using this is recommeneded only if set() and reset()
+   * Using this is recommended only if set() and reset()
    * are not being used in that parallel section/phase
    *
    * @param index Bit to check to see if set
    * @returns true if index is set
    */
   bool test(size_t index) const {
+    KATANA_ASSERT(index < num_bits);
```

Comment:
Why not just `assert()`? That would not be checked on release builds, which I think is desirable in this case.

Reply:
Does `KATANA_ASSERT` not get compiled out of release builds? I agree that this would be best as a debug-only check.

Reply:
```
//! Like assert but unconditionally executed
#define KATANA_ASSERT(cond, ...)      
```
So yes, it is for release asserts.

Reply:
Whoops, thanks for catching that.